### ISSUE TITLE
Failing is.numeric test

### ISSUE THREAD

**Issue by lennyxc** [max-alletsee/pricesensitivitymeter#19]
Trying to run psm_analysis() on a dataframe results in error "All price variables (toocheap, cheap, expensive, tooexpensive) must contain only numeric values\n(toocheap is also tolerated if all values are NA)"

When testing is.numeric(df$tex) for example, on all columns, it all returns true so unsure why this test is failing:

```
    if(ifelse(!is.numeric(data[, col_toocheap]), !all(is.na(data[, col_toocheap])), FALSE) | !is.numeric(data[, col_cheap]) |
       !is.numeric(data[, col_expensive]) | !is.numeric(data[, col_tooexpensive])) {
      stop("All price variables (toocheap, cheap, expensive, tooexpensive) must contain only numeric values\n(toocheap is also tolerated if all values are NA)")
    }
```

Tested on cran release as well as master branch in rstudio.cloud

**Comment by lennyxc** [comment]
Following up, R 3.5.3 works, but fails on R 4

**Comment by max-alletsee** [comment]
Hi @lennyxc ,
sorry for the late reply, I saw this just now. I just checked and this case was covered by unit testing, so it's strange to see that it fails in a real-life example. Could you maybe share some example code that you have used?
It would be interesting to understand the class of your data frame. Was it "data.frame" or something else like e.g. "tibble"?

**Comment by lennyxc** [comment]
@max-alletsee, No problem, I don't recall what steps I took to reduplicate or if I have the specific code anymore. However, I do recall I fixed the issue by using a previous version of R. I'll take a look to see if I have the history somewhere, but if I reach a dead-end, feel free to close!

**Comment by max-alletsee** [comment]
Hi @lennyxc ,
Thanks for your quick answer and appreciate that you search in your history. Totally understand that it may be very difficult after almost 6 months, just let me know if you can't find it anymore and I might close this. (Most functions are written using base-R, so if you solved it by going to a previous R version, chances are high that it's indeed an issue in the psm package and unrelated to changes in other packages.) If you can find the example, I'd be glad to check the root cause in the psm package. Thanks!

**Comment by lennyxc** [comment]
Found the old data and history, tested R 3.5.3, 3.5.3, and also 4.0.3 with pricesensitivitymeter 1.1.1. Not encountering to the error anymore, unsure why! For the sake of comprehensiveness, real life example was:

`library(pricesensitivitymeter)
library(readr)
r3v5 <- read_csv("data.csv")
output.psm.r3v5 <- psm_analysis(toocheap = r3v5$tec,cheap = r3v5$ch,expensive = r3v5$ex,tooexpensive = r3v5$tex
summary(output.psm.r3v5)`

**Comment by lennyxc** [comment]
Ah actually, I stand corrected on the above, I didn't encounter the error because I didn't change the workaround I used in: https://github.com/max-alletsee/pricesensitivitymeter/issues/19#issuecomment-684063763. I get the error in the OP after running these commands, regardless of version:

```
> library(readr)
> library(pricesensitivitymeter)
> b5 <- read_csv("Back Label_20200831_PSM_5.csv")
> output.psm.b5 <- psm_analysis(toocheap = "tch", cheap = "ch", expensive = "ex", tooexpensive = "tex", data = b5)
```

**Comment by max-alletsee** [comment]
Thanks for re-opening the issue and for sharing this example @lennyxc . I'm able to replicate your error and also figured out where this is coming from: subsetting in tibbles...

The package currently supports data input via data.frame and matrix objects. Could you maybe check if the following would ***not*** throw an error? (It worked for me with a test data set, but it would be great if it also works with "real-life" data.)

> library(readr)
> library(pricesensitivitymeter)
> b5 <- read_csv("data.csv")
> b5_df <- as.data.frame(b5)
> output.psm.b5 <- psm_analysis(toocheap = "tch", cheap = "ch", expensive = "ex", tooexpensive = "tex", data = b5_df)

If this works fine, I know the reason for the tibble issue: For data input, the package matches the column names from the function input to the data in a way that is not supported by tibble. :-\ It's basically the "neutral evil" [way described here](https://tibble.tidyverse.org/articles/wip/subset.html), as this allows to throw errors nicely if there is no match - but tibbles don't support this and always throw errors.

The good news is that I would probably need to re-work only [this beautiful line of spaghetti code](https://github.com/max-alletsee/pricesensitivitymeter/blob/master/R/psm_functions.R#L64) to fix everything... I will think about it on one of the next two weekeds.

If you could check if the quick fix in the code example helps, it would be amazing. Please keep it open nevertheless, I'd also like to fix the underlying issue.

**Comment by max-alletsee** [comment]
Note to myself: should now work with commit [a6672f59bf834d6af266139da179711614de9791](https://github.com/max-alletsee/pricesensitivitymeter/commit/a6672f59bf834d6af266139da179711614de9791)

**Comment by TobiasBenz** [comment]
Just in case: I ran into the same error and it worked using the declaration as.data.frame

**Comment by max-alletsee** [comment]
Hi @TobiasBenz ,
thanks for the comment, very helpful. After the update in this repo on January, the development version of the package should now also work nicely with tibbles.

If you have a few minutes, would you maybe be so kind to install the development version and try if it now indeed works with the tibble. (I'd love to have this real-life check with something other than my simple test dataset before sending it to Github...)

Here's the code that should install the development version from this repo:

    library(devtools)
    devtools::install_github("max-alletsee/pricesensitivitymeter")

Thanks,
Max

**Comment by TobiasBenz** [comment]
Hi Max,

sorry I cannot help atm as I am running a portable and thus old version of R. Hence I cannot install RTools. 

Best,
Tobias

**Comment by max-alletsee** [comment]
Hi Tobias,

ok, thanks anyway. I'm pretty sure that it's fixed, so I submitted the v1.2 to CRAN and will close this topic. Please re-open it if it's still an issue after updating the package to version 1.2 that will hopefully become available on CRAN soon.
